src/colmap/scene/reconstruction.cc

```diff
@@ -301,101 +302,80 @@ Sim3d Reconstruction::Normalize(const bool fixed_scale,
   const size_t num_elements = use_images ? NumRegImages() : points3D_.size();
   if (num_elements == 0) {
-    return std::make_tuple(Eigen::Vector3d(0, 0, 0),
-                           Eigen::Vector3d(0, 0, 0),
-                           Eigen::Vector3d(0, 0, 0));
+    return std::make_pair(
+        Eigen::AlignedBox3d(Eigen::Vector3d(0, 0, 0), Eigen::Vector3d(0, 0, 0)),
+        Eigen::Vector3d(0, 0, 0));
   }
 
   // Coordinates of image centers or point locations.
-  std::vector<float> coords_x;
-  std::vector<float> coords_y;
-  std::vector<float> coords_z;
+  std::vector<double> coords_x;
+  std::vector<double> coords_y;
+  std::vector<double> coords_z;
+  coords_x.reserve(num_elements);
+  coords_y.reserve(num_elements);
+  coords_z.reserve(num_elements);
   if (use_images) {
-    coords_x.reserve(NumRegImages());
-    coords_y.reserve(NumRegImages());
-    coords_z.reserve(NumRegImages());
-    for (const image_t im_id : RegImageIds()) {
-      const Eigen::Vector3d proj_center = Image(im_id).ProjectionCenter();
-      coords_x.push_back(static_cast<float>(proj_center(0)));
-      coords_y.push_back(static_cast<float>(proj_center(1)));
-      coords_z.push_back(static_cast<float>(proj_center(2)));
+    for (const image_t image_id : RegImageIds()) {
+      const Eigen::Vector3d proj_center = Image(image_id).ProjectionCenter();
+      coords_x.push_back(proj_center(0));
+      coords_y.push_back(proj_center(1));
+      coords_z.push_back(proj_center(2));
     }
   } else {
-    coords_x.reserve(points3D_.size());
-    coords_y.reserve(points3D_.size());
-    coords_z.reserve(points3D_.size());
     for (const auto& point3D : points3D_) {
-      coords_x.push_back(static_cast<float>(point3D.second.xyz(0)));
-      coords_y.push_back(static_cast<float>(point3D.second.xyz(1)));
-      coords_z.push_back(static_cast<float>(point3D.second.xyz(2)));
+      coords_x.push_back(point3D.second.xyz(0));
+      coords_y.push_back(point3D.second.xyz(1));
+      coords_z.push_back(point3D.second.xyz(2));
     }
   }
 
-  // Determine robust bounding box and mean.
-
-  std::sort(coords_x.begin(), coords_x.end());
-  std::sort(coords_y.begin(), coords_y.end());
-  std::sort(coords_z.begin(), coords_z.end());
-
-  const size_t P0 = static_cast<size_t>(
-      (coords_x.size() > 3) ? p0 * (coords_x.size() - 1) : 0);
-  const size_t P1 = static_cast<size_t>(
-      (coords_x.size() > 3) ? p1 * (coords_x.size() - 1) : coords_x.size() - 1);
-
-  const Eigen::Vector3d bbox_min(coords_x[P0], coords_y[P0], coords_z[P0]);
-  const Eigen::Vector3d bbox_max(coords_x[P1], coords_y[P1], coords_z[P1]);
-
-  Eigen::Vector3d mean_coord(0, 0, 0);
-  for (size_t i = P0; i <= P1; ++i) {
-    mean_coord(0) += coords_x[i];
-    mean_coord(1) += coords_y[i];
-    mean_coord(2) += coords_z[i];
-  }
-  mean_coord /= P1 - P0 + 1;
-
-  return std::make_tuple(bbox_min, bbox_max, mean_coord);
+  return ::colmap::ComputeBoundingBoxAndCentroid(min_percentile,
```

Comment:
Would it make sense to rename Reconstruction::ComputeBoundingBoxAndCentroid to avoid the name conflict?